src/visitors/hoist.js

```diff
@@ -209,7 +209,10 @@ module.exports = {
   CallExpression(path, asset) {
     let {callee, arguments: args} = path.node;
 
-    if (t.isIdentifier(callee, {name: '$parcel$exportWildcard'})) {
+    if (
+      t.isIdentifier(callee, {name: '$parcel$exportWildcard'}) ||
+      asset.package.sideEffects === false
+    ) {
       // This hints Uglify and Babel that this CallExpression does not have any side-effects.
       // This will make unsused CommonJS wildcards removed from the minified builds.
       path.addComment('leading', '#__PURE__');
```

Comment:
probably only want to add this to top-level calls right? e.g. not calls inside another function.

Reply:
Seems like that should be removed still with `sideEffects: false`. We know there are no side effects because of that flag, so removing the outer call should be safe. And since the inner one is inside a function called by the outer one, that should be safe too.

I was mostly concerned with marking calls inside function declarations as pure:

```javascript
function test() {
  return foo();
}
```

In that example, `foo` shouldn't be marked as pure since it is inside another function.

Maybe rather than checking if it's a top-level statement you could check the scope instead to make sure the call is not inside a function/class method scope? i.e. you could have a call inside a block scope (e.g. if statement) in the top-level scope that should still be marked as pure.